ci-operator/step-registry/hypershift/mce/power/create/ibmcloud-workers/lb/hypershift-mce-power-create-ibmcloud-workers-lb-commands.sh

```diff
@@ -0,0 +1,157 @@
+	TARGET_IP=$1
+	POOL_ID=$2
+	POOL_PORT=$3
+	ibmcloud is load-balancer-pool-member-create "${LB_ID}" "${POOL_ID}" "${POOL_PORT}" "${TARGET_IP}"
+	wait_lb_active 30
+}
+
+provision_backend_pool() {
+	POOL_NAME=$1
+	POOL_PORT=$2
+	# Create pool
+	POOL_ID=$(ibmcloud is load-balancer-pool-create "${POOL_NAME}" "${LB_ID}" round_robin tcp 5 2 2 tcp --health-monitor-port "${POOL_PORT}" --output JSON | jq -r '.id')
+	# attach servers
+	IFS=' ' read -r -a ips <<<"${IP_X86}"
+	for ip in "${ips[@]}"; do
+		attach_server $ip "${POOL_ID}" "${POOL_PORT}"
+	done
+	IFS=' ' read -r -a ips <<<"${IP_POWER}"
+	for ip in "${ips[@]}"; do
+		attach_server $ip "${POOL_ID}" "${POOL_PORT}"
+	done
+	# Create Frontend listener
+	ibmcloud is load-balancer-listener-create "${LB_ID}" --port "${POOL_PORT}" --protocol tcp --default-pool "${POOL_ID}"
+}
+
+provision_lb() {
+	# Create Load Balancer
+	LB_ID=$(ibmcloud is load-balancer-create "${LB_NAME}" public --vpc "${VPC_NAME}" --subnet "${VPC_SUBNET_ID}" --resource-group-name "${RESOURCE_GROUP_NAME}" --family application --output JSON | jq -r '.id')
+	wait_lb_active 300
+	echo "Load Balancer created!"
+	# Create Backend pools
+	provision_backend_pool "http" "80"
+	wait_lb_active 30
+	provision_backend_pool "https" "443"
+}
+
+update_dns_records() {
+	HOSTED_CLUSTER_API_SERVER=$(oc get service kube-apiserver -n ${HOSTED_CONTROL_PLANE_NAMESPACE} -o json | jq -r '.status.loadBalancer.ingress[].hostname')
+	IP_ADDRESSES=$(cat ${SHARED_DIR}/ippoweraddr)
+
+	# Creating dns records in ibmcloud cis service for agents to reach hosted cluster
+	ibmcloud cis dns-record-create ${CIS_DOMAIN_ID} --type CNAME --name "api.${HOSTED_CLUSTER_NAME}" --content "${HOSTED_CLUSTER_API_SERVER}"
+	ibmcloud cis dns-record-create ${CIS_DOMAIN_ID} --type CNAME --name "api-int.${HOSTED_CLUSTER_NAME}" --content "${HOSTED_CLUSTER_API_SERVER}"
+
+	if [ ${USE_GLB} == "yes" ]; then
+		origins=$(sed 's/,$//' "${SHARED_DIR}/origins")
+		origin_pools_json="{\"name\": \"${HOSTED_CLUSTER_NAME}\", \"origins\": [${origins}]}"
+
+		pool_id=$(ibmcloud cis glb-pool-create -i ${CIS_INSTANCE} --json "${origin_pools_json}" --output json | jq -r '.id')
+
+		lb_name="${HOSTED_CLUSTER_NAME}.${HYPERSHIFT_BASE_DOMAIN}"
+		lb_payload="{\"name\": \"${lb_name}\",\"fallback_pool\": \"${pool_id}\",\"default_pools\": [\"${pool_id}\"]}"
+
+		ibmcloud cis glb-create ${CIS_DOMAIN_ID} -i ${CIS_INSTANCE} --json "${lb_payload}"
+
+		# Creating dns record for ingress
+		echo "$(date) Creating dns record for ingress"
+		ibmcloud cis dns-record-create ${CIS_DOMAIN_ID} --type CNAME --name "*.apps.${HOSTED_CLUSTER_NAME}" --content "${lb_name}"
+	elif [ ${USE_GLB} == "no" ]; then
+		echo "$(date) GLB not created, so assigning first node's ip to ingress dns record"
+		ibmcloud cis dns-record-create ${CIS_DOMAIN_ID} --type A --name "*.apps.${HOSTED_CLUSTER_NAME}" --content "${IP_ADDRESSES[0]}"
+	else
+		echo "DNS record entry for *.apps.${HOSTED_CLUSTER_NAME} not added."
+	fi
+}
+
+main() {
```

Comment:
I can see LB is created for power inside `update_dns_records` itself. Can we refactor by moving that to provision_lb?

Reply:
Done